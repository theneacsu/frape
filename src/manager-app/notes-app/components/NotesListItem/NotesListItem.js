import React from 'react'
import { Link } from 'react-router-dom'
import styles from './NotesListItem.module.css'

const NotesListItem = props => (
  <div className={styles.div}>
    <div>
      <Link to={`/notes/${props.id}`} className={styles.a}>
        <h1 className={styles.h1}>{props.title}</h1>
      </Link>
    </div>
    <Link to={`/notes/${props.id}`} className={styles.a}>
      <p className={styles.p}>{props.body}</p>
    </Link>
  </div>
)

export default NotesListItem
