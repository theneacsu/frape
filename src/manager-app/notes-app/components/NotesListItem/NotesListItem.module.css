.div {
  margin: 0 0 1.5rem 0;
  width: 70%;
  justify-content: center;
  padding: 0;
}

.a {
  text-decoration: none;
}

.h1 {
  margin: 0;
  border-top-left-radius: 1rem;
  border-top-left-radius: 1rem;
  border-top-right-radius: 1rem;
  font-size: 1.5rem;
  text-align: center;
  color: white;
  background-color: #aa76f8;
  padding: 1rem;
  word-wrap: break-word;
}

.p {
  margin: 0;
  word-wrap: break-word;
  font-size: 1.5rem;
  padding: 1rem;
  text-align: center;
  color: #47667c;
  border: 1px solid #aa76f8;
  border-bottom-left-radius: 1.8rem;
  border-bottom-right-radius: 1.8rem;
}

@media (min-width: 355px) {
  .div {
    width: 45%;
    margin-left: 0.6rem;
    margin-right: 0.8rem;
  }
}

@media (min-width: 450px) {
  .div {
    width: 45%;
  }
}

@media (min-width: 600px) {
  .h1 {
    font-size: 1.8rem;
  }
}

@media (min-width: 750px) {
  .div {
    width: 30.5%;
  }
}
