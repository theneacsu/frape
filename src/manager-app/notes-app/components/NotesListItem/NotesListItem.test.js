import React from 'react'
import { shallow } from 'enzyme'
import NotesListItem from './NotesListItem'

test('Should render the NotesListItem component correctly', () => {
  const note = {
    id: '123abc',
    text: 'Go to store',
    body: 'Buy food'
  }
  const wrapper = shallow(<NotesListItem {...note} />)
  expect(wrapper).toMatchSnapshot()
})
