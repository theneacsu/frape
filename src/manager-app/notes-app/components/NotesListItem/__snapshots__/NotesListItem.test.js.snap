// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the NotesListItem component correctly 1`] = `
<div
  className="div"
>
  <div>
    <Link
      className="a"
      replace={false}
      to="/notes/123abc"
    >
      <h1
        className="h1"
      />
    </Link>
  </div>
  <Link
    className="a"
    replace={false}
    to="/notes/123abc"
  >
    <p
      className="p"
    >
      Buy food
    </p>
  </Link>
</div>
`;
